runtime/src/main/java/org/corfudb/runtime/collections/UFOTxnBase.java

```diff
@@ -0,0 +1,369 @@
+        return putRecord(table, key, value, metadata, false);
+    }
+
+    @Nonnull
+    public <K extends Message, V extends Message, M extends Message>
+    UFOTxnBase putRecord(@Nonnull String tableName,
+                         @Nonnull final K key,
+                         @Nonnull final V value,
+                         @Nullable final M metadata) {
+        return putRecord(this.txnContext.getTable(tableName), key, value, metadata, false);
+    }
+
+    @Nonnull
+    public <K extends Message, V extends Message, M extends Message>
+    UFOTxnBase putRecord(@Nonnull String tableName,
+                         @Nonnull final K key,
+                         @Nonnull final V value,
+                         @Nullable final M metadata,
+                         boolean forceSetMetadata) {
+        return putRecord(this.txnContext.getTable(tableName), key, value, metadata, forceSetMetadata);
+    }
+
+    /**
+     * put the value on the specified key create record if it does not exist.
+     *
+     * @param table    Table object to perform the create/update on.
+     * @param key      Key of the record.
+     * @param value    Value or payload of the record.
+     * @param metadata Metadata associated with the record.
+     * @param forceSetMetadata if specified, metadata will not be validated, just set
+     * @param <K>      Type of Key.
+     * @param <V>      Type of Value.
+     * @param <M>      Type of Metadata.
+     * @return TxnContext instance.
+     */
+    @Nonnull
+    public <K extends Message, V extends Message, M extends Message>
+    UFOTxnBase putRecord(@Nonnull Table<K, V, M> table,
+                         @Nonnull final K key,
+                         @Nonnull final V value,
+                         @Nullable final M metadata,
+                         boolean forceSetMetadata) {
+
+        if (forceSetMetadata || metadata == null) {
+            this.txnContext = txnContext.put(table, key, value, null);
+            return this;
+        }
+
+        this.txnContext = txnContext.merge(table, key,
+                this::fixUpMetadata, new CorfuRecord<>(value, metadata));
+        return this;
+    }
+
+    /**
+     * Core logic for handling metadata modifications on mutations
+     * @param oldRecord - The previous fetched record that exists in the table.
+     * @param deltaUpdate - New record that will be inserted/updated.
+     * @param <K> - type of the key
+     * @param <V> - type of the value or payload
+     * @param <M> - type of the metadata
+     * @return the merged record that will be inserted into the table.
+     */
+    private <K extends Message, V extends Message, M extends Message>
+    CorfuRecord<V, M> fixUpMetadata(CorfuRecord<V, M> oldRecord, CorfuRecord<V, M> deltaUpdate) {
+        M deltaMetadata = deltaUpdate.getMetadata();
+            final Message.Builder builder = deltaMetadata.toBuilder();
+            long currentTime = System.currentTimeMillis();
```

Comment:
Need to format the code, the indent is somehow mis-aligned

Reply:
fixed.